test/MorseL.Sockets.Tests/MorseLHttpMiddlewareTests.cs

```diff
@@ -0,0 +1,165 @@
+        {
+            return new MorseLHttpMiddleware(
+                Mocker.ServiceProviderMock.Object.GetRequiredService<ILogger<MorseLHttpMiddleware>>(),
+                RequestDelegateMock.Object.Next,
+                typeof(HubWebSocketHandler<MiddlewareTestHub>),
+                cts);
+        }
+
+        [Fact]
+        public async Task NonWebSocketRequest_InvokesNextRequestDelegate()
+        {
+            var sut = CreateMorseLHttpMiddleware();
+
+            Mocker.WebSocketManagerMock.Setup(m => m.IsWebSocketRequest).Returns(false);
+
+            await sut.Invoke(Mocker.HttpContextMock.Object);
+
+            RequestDelegateMock.Verify(m => m.Next(It.Is<HttpContext>(hc => hc == Mocker.HttpContextMock.Object)), Times.Once);
+        }
+
+        [Fact]
+        public async Task WebSocketRequest_DoesNotInvokeNextRequestDelegate()
+        {
+            var sut = CreateMorseLHttpMiddleware();
+
+            await sut.Invoke(Mocker.HttpContextMock.Object);
+
+            RequestDelegateMock.Verify(m => m.Next(It.IsAny<HttpContext>()), Times.Never);
+        }
+
+        [Fact]
+        public async Task WebSocketRequest_ShouldBeAddedToConnectionManager()
+        {
+            var sut = CreateMorseLHttpMiddleware();
+
+            await sut.Invoke(Mocker.HttpContextMock.Object);
+
+            Mocker.WebSocketConnectionManagerMock.Verify(m => m.AddConnection(It.IsAny<IChannel>()), Times.Once);
+        }
+
+        [Fact]
+        public async Task WebSocketRequest_ShouldBeRemovedFromConnectionManager_IfSocketIsNotOpen()
+        {
+            Mocker.WebSocketMock.Setup(m => m.State).Returns(WebSocketState.None);
+
+            var sut = CreateMorseLHttpMiddleware();
+
+            await sut.Invoke(Mocker.HttpContextMock.Object);
+
+            Mocker.WebSocketConnectionManagerMock
+                .Verify(m => m.RemoveConnection(It.Is<string>(s => s == Mocker.ConnectionId)), Times.Once);
+        }
+
+        [Fact]
+        public async Task WebSocketRequest_ShouldBeRemovedFromConnectionManager_IfCancellationIsRequested()
+        {
+            var cts = new CancellationTokenSource();
+            cts.Cancel();
+
+            var sut = CreateMorseLHttpMiddleware(cts);
+
+            await sut.Invoke(Mocker.HttpContextMock.Object);
+
+            Mocker.WebSocketConnectionManagerMock
+                .Verify(m => m.RemoveConnection(It.Is<string>(s => s == Mocker.ConnectionId)), Times.Once);
+        }
+
+        [Fact]
+        public async Task WebSocketRequest_ShouldBeRemovedFromConnectionManager_IfExceptionThrownInReceiveLoop()
+        {
+            Mocker.WebSocketMock.Setup(m => m.State).Returns(WebSocketState.Open);
+            Mocker.WebSocketMock.Setup(m => m.ReceiveAsync(It.IsAny<ArraySegment<byte>>(), It.IsAny<CancellationToken>()))
+                .Throws<Exception>();
+
+            var sut = CreateMorseLHttpMiddleware();
```

Comment:
What is sut?

Reply:
"System Under Test" - I have more explanation on it in [this comment](https://github.com/ryanwersal/MorseL/pull/44#discussion_r226326389).